src/module.c

```diff
@@ -10712,6 +10707,22 @@ int moduleUnload(sds name) {
     return C_OK;
 }
 
+/* This function will check all modules in order to unload
+ * the modules with `unload_asap` is set. */
+void moduleHandleAsyncUnload(void) {
+    dictIterator *di = dictGetSafeIterator(modules);
+    dictEntry *de;
+
+    while ((de = dictNext(di)) != NULL) {
+        struct RedisModule *module = dictGetVal(de);
+        if (module->unload_asap) {
+            module->unload_asap = 0;
+            moduleUnload(module);
```

Comment:
when this fails, let's at least log `errno` to the log file.
maybe log it on success too..

Reply:
Regardless of success or failure, `moduleUnload` has logged to the log file.

Reply:
Ohh, Indeed, I will do it.

Reply:
Done it.